<launch>
  <!-- Launch file for velodyne16 VLP-16 LiDAR -->

    <arg name="rviz" default="true" />

    <rosparam command="load" file="$(find lidar_localization)/config/velodyne.yaml" />

    <node pkg="lidar_localization" type="lidar_localization" name="lidar_localization" output="screen" /> 
    
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
		<param name="resolution" value="0.15" />
		
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<param name="frame_id" type="string" value="map" />
		
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="60.0" />

		<param name="pointcloud_min_z" value="-5.0" />
		<param name="pointcloud_max_z" value="5.5" />
		<param name="occupancy_min_z" value="-0.5" />
		<param name="occupancy_max_z" value="0.5" />
		<param name="sensor_model/hit" value="0.7" />
		<param name="sensor_model/miss" value="0.4" />		
		
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="/cloud_body" />	
	  </node>
	  

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find lidar_localization)/rviz/lidar_local.rviz" />
    </group>

</launch>